x-pack/plugins/apm/public/store/urlParams.ts

```diff
@@ -19,6 +20,27 @@ import { IReduxState } from './rootReducer';
 
 // ACTION TYPES
 export const TIMEPICKER_UPDATE = 'TIMEPICKER_UPDATE';
+export const TIMEPICKER_DEFAULTS = {
+  rangeFrom: 'now-24h',
+  rangeTo: 'now',
+  rangePaused: 'true',
+  refreshInterval: '0'
+};
+
+function calculateTimePickerDefaults() {
+  const parsed = {
+    from: datemath.parse(TIMEPICKER_DEFAULTS.rangeFrom),
+    // roundUp: true is required for the quick select relative date values to work properly
+    to: datemath.parse(TIMEPICKER_DEFAULTS.rangeTo, { roundUp: true })
+  };
+
+  return {
+    start: parsed.from ? parsed.from.toISOString() : undefined,
+    end: parsed.to ? parsed.to.toISOString() : undefined
+  };
+}
+
+const INITIAL_STATE = calculateTimePickerDefaults();
```

Comment:
nit: since the input values are constants it's not necessary to check the parsed result, and something a bit shorter might suffice:

```js
const INITIAL_STATE = {
    start: datemath.parse(TIMEPICKER_DEFAULTS.rangeFrom).toISOString(),
    end: datemath.parse(TIMEPICKER_DEFAULTS.rangeTo, { roundUp: true }).toISOString()
}
```

Reply:
`datemath.parse` can return undefined so you have to guard against undefined to satisfy the TS gods.

Reply:
Ah.. makes sense.

Reply:
(although in this case, it would be better if `datemath.parse` would throw on invalid input, instead of return `undefined`)

Reply:
Agreed, I've had to do this check in a lot of places for what seems like no reason.